source/hud.lua

```diff
@@ -141,18 +141,22 @@ local function drawShopMenu()
 		Assets.setFont("font20")
 
 		-- Create List of available modules
-		for _, module in pairs(Modules) do
-			local string = "%s. Buy %s - $%s \n"
-			itemListString = string.format(string, module.id, module.name, module.cost)
-			-- Draw list of modules
-			local color = {1, 1, 1, 1}
-			local y = SCREEN_HEIGHT * 0.33
-			if Lander.hasUpgrade(LANDERS[1], module) then
-				color = {.8, .1, .1, .5}
+		local y = SCREEN_HEIGHT * 0.33
+
+		for k, module in pairs(Modules) do
+			if module.allowed == nil or module.allowed == true then
```

Comment:
`if module.allowed == nil or module.allowed == true then` is over complicated imo. We should just add this variable for every module and do `if module.allowed then` instead.

Also I'd not remove the entry from the list  but instead just draw it gray or transparent or something so that players still can see that there is something they can't use for whatever reason atm. That would be better then removing it and worst case ppl wouldn't know that there is content which is just locked behind some setting or dependency.